src/templates.rs

```diff
@@ -0,0 +1,37 @@
+use log::error;
+use std::fs;
+use std::path::PathBuf;
+
+/// Creates the `templates/` folder and base template file.
+pub fn initialize_templates(output_folder: &PathBuf) {
+    let templates_path = output_folder.join("templates");
+    if let Err(e) = fs::create_dir_all(&templates_path) {
+        error!("Failed to create templates directory: {}", e);
+        return;
+    }
+
+    let base_template_path = templates_path.join("base.html");
+    if let Err(e) = fs::write(base_template_path, "<!-- Base HTML template -->") {
+        error!("Failed to create base template: {}", e);
+    }
```

Comment:
`output_folder` should actually be the `inout_folder` in this case,
here we are writing templates, the output_folder is where we write the rendered files.

We want to write all the embedded templates with its contents.

Here you can use https://github.com/rochacbruno/marmite/blob/89841683bbc469ce1ac6d65a475ac375ec91b3ef/src/embedded.rs#L16

Something like

```rust
for name in Templates::iter() {
    let template = Templates::get(name.as_ref()).unwrap();
    let template_str = std::str::from_utf8(template.data.as_ref()).unwrap();
    # here you write to `templates_path.join(&name)` the contents of template_str
}
```

Reply:
Instead of writing to output_folder, initialize_templates now writes to input_folder. The function creates the templates directory within input_folder and iterates through all embedded templates using Templates::iter()